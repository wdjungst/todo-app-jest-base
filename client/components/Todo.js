import React from 'react';

const Todo = ({ _id, text, completed, onUpdate, onDelete }) => (
  <li 
    className="collection-item" 
    style={ completed ? 
      { cursor: 'pointer', textDecoration: 'line-through', color: 'grey' } : {cursor: 'pointer' }
    }
    onClick={ () => onUpdate(_id, !completed) }
  >
    <div>
      {text}
      <a
        className="secondary-content"
        onClick={ e => { 
          e.preventDefault()
          onDelete(_id) 
        }}
      >
        Delete
      </a>
    </div>
   </li>
)

export default Todo;
