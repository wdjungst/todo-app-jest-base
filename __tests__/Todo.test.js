import React from 'react';
import Todo from '../client/components/Todo';
import renderer from 'react-test-renderer';

describe('Todo', () => {
  test('Component is rendered', () => {
    const onUpdate = (id, completed) => {
      return null
    }
  
    const onDelete = (id) => {
      return null
    }
  
    const component = renderer.create(
      <Todo onDelete={onDelete} onUpdate={onUpdate} _id={1} text="Buy Milk" completed={false} />
    );
  
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    tree.props.onClick();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('Component is rendered with completed style', () => {
    const component = renderer.create(
      <Todo _id={1} text="Buy Milk" completed={true} />
    )
  
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
