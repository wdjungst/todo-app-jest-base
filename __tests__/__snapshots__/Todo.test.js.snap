exports[`Todo Component is rendered 1`] = `
<li
  className="collection-item"
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }>
  <div>
    Buy Milk
    <a
      className="secondary-content"
      onClick={[Function]}>
      Delete
    </a>
  </div>
</li>
`;

exports[`Todo Component is rendered 2`] = `
<li
  className="collection-item"
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }>
  <div>
    Buy Milk
    <a
      className="secondary-content"
      onClick={[Function]}>
      Delete
    </a>
  </div>
</li>
`;

exports[`Todo Component is rendered with completed style 1`] = `
<li
  className="collection-item"
  onClick={[Function]}
  style={
    Object {
      "color": "grey",
      "cursor": "pointer",
      "textDecoration": "line-through",
    }
  }>
  <div>
    Buy Milk
    <a
      className="secondary-content"
      onClick={[Function]}>
      Delete
    </a>
  </div>
</li>
`;

exports[`test Component is rendered 1`] = `
<li
  className="collection-item"
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }>
  <div>
    Buy Milk
    <a
      className="secondary-content"
      onClick={[Function]}>
      Delete
    </a>
  </div>
</li>
`;

exports[`test Component is rendered 2`] = `
<li
  className="collection-item"
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }>
  <div>
    Buy Milk
    <a
      className="secondary-content"
      onClick={[Function]}>
      Delete
    </a>
  </div>
</li>
`;

exports[`test Component is rendered with completed style 1`] = `
<li
  className="collection-item"
  onClick={[Function]}
  style={
    Object {
      "color": "grey",
      "cursor": "pointer",
      "textDecoration": "line-through",
    }
  }>
  <div>
    Buy Milk
    <a
      className="secondary-content"
      onClick={[Function]}>
      Delete
    </a>
  </div>
</li>
`;
